backend/Dockerfile

```diff
@@ -0,0 +1,11 @@
+FROM golang:latest
+
+WORKDIR /app
+
+COPY ./backend /app
+COPY ./certs /certs
+
+RUN go mod download
+RUN go get github.com/githubnemo/CompileDaemon
```

Comment:
Why not just build the application and start it?

Reply:
What's happening here is that I run the app with `CompileDaemon`, and then in `docker-compose-dev.yml` I mount the `backend/` directory onto this container. `CompileDaemon` then detects whenever I make any changes to any `backend/*.go` files and recompiles/serves the app, effectively giving me a hot-reload server for development. 

I think the confusion may stem from the fact that this Dockerfile is only used for the development setup? If that's the case then I could clarify by renaming it `dev.Dockerfile`

Reply:
That's fine, makes sense now.